src/prefect/infrastructure/docker.py

```diff
@@ -162,18 +169,39 @@ class DockerContainer(Infrastructure):
     necessary and the API is connectable while bound to localhost.
     """
 
-    image: str = Field(default_factory=get_prefect_image_name)
-    image_pull_policy: ImagePullPolicy = None
+    type: Literal["docker-container"] = Field(
+        "docker-container", description="The type of infrastructure"
+    )
+    image: str = Field(
+        description="Tag of a Docker image to use. Defaults to the Prefect image.",
+        default_factory=get_prefect_image_name,
+    )
+    image_pull_policy: ImagePullPolicy = Field(
+        None, description="Specifies if the image should be pulled"
+    )
     image_registry: Optional[DockerRegistry] = None
-    networks: List[str] = Field(default_factory=list)
-    network_mode: str = None
-    auto_remove: bool = False
-    volumes: List[str] = Field(default_factory=list)
-    stream_output: bool = True
+    networks: List[str] = Field(
+        default_factory=list,
+        description="A list of strings specifying Docker networks to connect the container to",
+    )
+    network_mode: Optional[str] = Field(
+        None,
+        description="The network mode for the created container (e.g. host, bridge). If 'networks' is set, this cannot be set.",
+    )
+    auto_remove: bool = Field(
+        False, description="If set, the container will be removed on completion"
```

Comment:
I think all of these should end in periods consistently per @tpdorsey's style.

Reply:
Makes sense. Added an assertion to the field description test and all descriptions end in periods.